Documentation/workflow/EventPipeDeveloperGuide.md

```diff
@@ -0,0 +1,105 @@
+
+CoreCLR (the EventPipes are built-in feature of CoreCLR):
+
+* [src/vm/eventpipe.cpp](../../src/vm/eventpipe.cpp)
+* [src/vm/eventpipebuffer.cpp](../../src/vm/eventpipebuffer.cpp)
+* [src/vm/eventpipebuffermanager.cpp](../../src/vm/eventpipebuffermanager.cpp)
+* [src/vm/eventpipeconfiguration.cpp](../../src/vm/eventpipeconfiguration.cpp)
+* [src/vm/eventpipeevent.cpp](../../src/vm/eventpipeevent.cpp)
+* [src/vm/eventpipeeventinstance.cpp](../../src/vm/eventpipeeventinstance.cpp)
+* [src/vm/eventpipefile.cpp](../../src/vm/eventpipefile.cpp) - creates and writes to Event Pipe file. It also writes the file header and maintains the forward references.
+* [src/vm/eventpipejsonfile.cpp](../../src/vm/eventpipejsonfile.cpp) - allows to use JSON instead of binary file format. Used only for debugging purposes.
+* [src/vm/eventpipeprovider.cpp](../../src/vm/eventpipeprovider.cpp)
+* [src/vm/fastserializer.cpp](../../src/vm/fastserializer.cpp) - fast binary serializer used by `EventPipeFile` to store data
+* [src/scripts/genEventPipe.py](../../src/scripts/genEventPipe.py) - Python script run at build time, generates all event definitions
+* [tests/src/tracing/eventpipetrace/EventPipeTrace.cs](../../tests/src/tracing/eventpipetrace/EventPipeTrace.cs) - Tests. As of today (18th of January 2018) these testes are not run by the CoreCLR CI.
+
+PerfView repository is the home for `Microsoft.Diagnostics.Tracing.TraceEvent` Library, which provide client classes that allow to consume Event Pipe:
+
+* [src/TraceEvent/EventPipe/EventPipeEventSourceFactory.cs](https://github.com/Microsoft/perfview/blob/master/src/TraceEvent/EventPipe/EventPipeEventSourceFactory.cs) - factory which parses the header of Event Pipe file and creates `EventPipeEventSource`
+* [src/TraceEvent/EventPipe/EventPipeEventSource.cs](https://github.com/Microsoft/perfview/blob/master/src/TraceEvent/EventPipe/EventPipeEventSource.cs) - processes all events information using `EventPipeTraceEventParser`
+* [src/TraceEvent/EventPipe/EventPipeTraceEventParser.cs](https://github.com/Microsoft/perfview/blob/master/src/TraceEvent/EventPipe/EventPipeTraceEventParser.cs) - parser responsible for the deserialization of data serialized by `FastSerializer` in CoreCLR
+* [src/TraceEvent/TraceEvent.Tests/EventPipeParsing.cs](https://github.com/Microsoft/perfview/blob/master/src/TraceEvent/TraceEvent.Tests/EventPipeParsing.cs) - tests
+
+## How to build the code
+
+### CoreCLR
+
+[Here](https://github.com/dotnet/coreclr#building-the-repository) you can find out how to build CoreCLR. 
+
+**Important:**  As of today (18th of January 2018) Event Pipe is disabled by default for Windows build of CoreCLR. To enable it, you need to set `FeaturePerfTracing` to `true` in the [clr.coreclr.props](../../clr.coreclr.props) file.
```

Comment:
This may change very shortly. @brianrob was the only change needed to enable the managed interface for EventPipe simply to remove the `#if`?